ethportal-peertest/src/scenarios/validation.rs

```diff
@@ -0,0 +1,117 @@
+use ethportal_api::types::enr::Enr;
+use ethportal_api::types::portal::ContentInfo;
+use ethportal_api::{
+    jsonrpsee::async_client::Client, HistoryContentKey, HistoryContentValue,
+    HistoryNetworkApiClient,
+};
+use serde_json::json;
+use std::str::FromStr;
+use tracing::info;
+
+use crate::{
+    constants::{
+        BLOCK_BODY_CONTENT_KEY, BLOCK_BODY_CONTENT_VALUE, HEADER_WITH_PROOF_CONTENT_KEY,
+        HEADER_WITH_PROOF_CONTENT_VALUE, RECEIPTS_CONTENT_KEY, RECEIPTS_CONTENT_VALUE,
+    },
+    Peertest,
+};
+
+pub async fn test_validate_pre_merge_header_with_proof_bb_receipts(
+    peertest: &Peertest,
+    target: &Client,
+) {
+    info!("Testing validating a pre-merge header-with-proof, bb, & receipts");
+
+    // store header_with_proof
+    let header_with_proof_content_key: HistoryContentKey =
+        serde_json::from_value(json!(HEADER_WITH_PROOF_CONTENT_KEY)).unwrap();
+    let header_with_proof_content_value: HistoryContentValue =
+        serde_json::from_value(json!(HEADER_WITH_PROOF_CONTENT_VALUE)).unwrap();
+
+    let store_result = peertest
+        .bootnode
+        .ipc_client
+        .store(
+            header_with_proof_content_key.clone(),
+            header_with_proof_content_value.clone(),
+        )
+        .await
+        .unwrap();
+
+    assert!(store_result);
+
+    // calling find_content since it only returns the found data if validation was successful
+    let result = target
+        .find_content(
+            Enr::from_str(&peertest.bootnode.enr.to_base64()).unwrap(),
+            header_with_proof_content_key.clone(),
+        )
+        .await
+        .unwrap();
+
+    match result {
+        ContentInfo::Content { content } => assert_eq!(content, header_with_proof_content_value),
+        _ => unreachable!(),
+    }
+
+    // store block body
```

Comment:
I'd split this test up into three different tests, one for hwp, one for bb, one for receipts. If we run into a failure, it makes it easier to understand which stage went wrong (eg. headers & bodies are working but receipts are failing)

Reply:
resolved